autogen/paths.py

```diff
@@ -15,12 +15,13 @@
 from __future__ import absolute_import, unicode_literals
 
 import os
+import shutils
+
 from glob import glob
-from distutils.spawn import find_executable
 
 
-# find_executable() returns None if nothing was found
-gppath = find_executable("gp")
+gppath = shutil.which("gp")
+
```

Comment:
I don't know what's "shutil" or "shutils" but I note that you use two different names.

Also: I didn't need this patch to successfully build cypari2 with python 3.12. What I PR is good enough (as in: I can build and fully test sagemath with it). If this is a problem, is not catched by the testsuite of cypari2 or sagemath. I know distutils is not available anymore, so maybe that means this file is not used at all? It's definitely not tested (as all tests pass for me IIRC).

PS: I find it intriguing the asymmetry that some sage developers can modify my PRs but I can't modify any PR not authored by me. Granted, that was a useful feature of trac, but it's not a standard feature of gh afaik. I don't mind you changing my PRs, just something to think about. I'm glad that you are working on getting releases for cypari2 (and the other pkgs we need).

Reply:
> I find it intriguing the asymmetry that some sage developers can modify my PRs but I can't modify any PR not authored by me.

One needs to be in the Maintainer role for the project. https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization
I am not sure if there is a clear protocol how and when to elevate developers to this role, but an email to sagemath-admins would probably help.

Reply:
Also, as PR author one can uncheck the box "Maintainers are allowed to edit this pull request." to remove unwelcome interference from maintainers.

Reply:
I've given sagemath/triage team (with @tornaria there) same rights here (and on few other sagemath/ repos) as on the main Sage repo. Hopefully this gives more rights  to fix stuff.

Reply:
@tornaria - [shutil](https://docs.python.org/3/library/shutil.html)